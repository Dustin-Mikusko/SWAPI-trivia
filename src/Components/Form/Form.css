/* * {
  border: 1px solid red;
} */

form {
  background-image: url('../../images/background-stars.jpeg') ;
  color: #fff;
  display: flex;
  justify-content: center;
  align-items: center;
  flex-direction: column;
  height: 100vh
}

img {
  width: 40%;
  height: 50%
}

input {
  height: 1.5em;
  width: 20%;
  margin: .5em;
  border-radius: 10px;
  font-size: 2em;
}

select {
  height: 2em;
  width: 20%;
  margin: .5em;
  border-radius: 10px;
  font-size: 1.5em;
}

button {
  height: 2em;
  width: 20%;
  margin: 1.5em;
  border-radius: 10px;
  font-size: 1.75em;
  background-color: #E8C539;
  font-weight: bold;
}

.error-border {
  border: 3px solid red;
}
