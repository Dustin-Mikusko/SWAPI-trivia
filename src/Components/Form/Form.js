import React, { Component } from 'react';
import './Form.css';
import starWarsLogo from '../../images/star-wars-logo.jpeg';
import { Link, Redirect } from 'react-router-dom';

class Form extends Component {
  constructor(props) {
    super(props);
    this.state = {
      name: '',
      quote: '',
      rank: '',
      error: {
<<<<<<< HEAD
          name: false,
          quote: false,
          rank: false,
        }
=======
        name: false,
        quote: false,
        rank: false
      },
>>>>>>> 652009c48336b8646b9653720b77bd9ee2f8fb9e
    }
  }

  handleChange = event => {
    this.setState({ [event.target.name]: event.target.value });
  }

  clearInputs = () => {
    this.setState({ name: '', quote: '', rank: '' });
  }

  submitUser = () => {
<<<<<<< HEAD
      const { addUser } = this.props;
      let user = {
        name: this.state.name,
        quote: this.state.quote,
        rank: this.state.rank,
        favoriteCharacters: [],
        loggedIn: true
      }
      addUser(user);
    }

  checkInputs = () => {
    let errorProperty = this.state.error;
    // let errorProperty2 = {...this.state.error};
    console.log(1, errorProperty);
    // console.log(2, errorProperty2);
=======
    const { addUser } = this.props;
    let user = {
      name: this.state.name,
      quote: this.state.quote,
      rank: this.state.rank,
      favoriteCharacters: [],
      loggedIn: true
    }
    addUser(user);
  }

  checkInputs = () => {
    let errorProperty = this.state.error;
    console.log(1, errorProperty);
>>>>>>> 652009c48336b8646b9653720b77bd9ee2f8fb9e
    Object.keys(errorProperty).forEach(key => {
      if (!this.state[key]) {
        errorProperty[key] = true;
        this.setState({error: errorProperty});
      } else {
        errorProperty[key] = false;
        this.setState({error: errorProperty})
      }
    });
    console.log(3, {...errorProperty});
<<<<<<< HEAD
    // console.log(4, errorProperty2);
    this.checkReady();
  }
  
=======
    this.checkReady();
  } 
>>>>>>> 652009c48336b8646b9653720b77bd9ee2f8fb9e
  checkReady = () => {
    let error = this.state.error;
    return !error.name && !error.quote && !error.rank ? this.submitUser() : console.log('nope, cann\'t do it');
  }


  render() {
    const nameErrorClass = this.state.error.name ? 'error-border' : '';
    const quoteErrorClass = this.state.error.quote ? 'error-border' : '';
    const rankErrorClass = this.state.error.rank ? 'error-border' : '';
    return (
      <form>
      <img src={starWarsLogo} alt='Star Wars logo' />
       <input
          className={nameErrorClass}
          type="text"
          placeholder=" Name"
          name="name"
          value={this.state.name}
          onChange={event => this.handleChange(event)}
        />
        <input
          className={quoteErrorClass}
          type="text"
          placeholder=" Favorite Star Wars Quote"
          name="quote"
          value={this.state.quote}
          onChange={event => this.handleChange(event)}
        />
        <select
          className={rankErrorClass}
          name="rank"
          value={this.state.rank}
          onChange={event => this.handleChange(event)}
          >
            <option value=''>--Choose Level--</option>
            <option value="beginner">What Is Star Wars??</option>
            <option value="intermediate">Padawan</option>
            <option value="expert">The Force is Strong With This One...</option>
        </select>
        <button type="button" onClick={this.checkInputs}>Enter, You Will</button>

      </form>
    )
  }
}

export default Form;
