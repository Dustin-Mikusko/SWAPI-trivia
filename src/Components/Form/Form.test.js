import React from 'react';
import Form from './Form';
import { shallow } from 'enzyme';

describe('Form', () => {
  let wrapper;
  let mockAddUser;

  beforeEach(() => {
    mockAddUser = jest.fn();
    wrapper = shallow(<Form addUser={mockAddUser}/>)
  });

  it('should match the snapshot with all the data passed in correctly', () => {
    const wrapper = shallow(<Form 
      addUser={jest.fn()}
    />);

    expect(wrapper).toMatchSnapshot();
  });

  it('should update state when handleChange is called', () => {
    const mockEvent = { target: {name: 'name', value: 'Tom Brady'} };
    const expected = 'Tom Brady';

    wrapper.instance().handleChange(mockEvent);

    expect(wrapper.state('name')).toEqual(expected);
  });

  it('should clear the clear the name, quote, and rank values in state when clearInputs is called', () => {
    const expected = {
      name: '',
      quote: '',
      rank: ''
    }

    wrapper.setState({
      name: 'Tom Brady',
      quote: 'My favorite ring is the next one.',
      rank: 'beginner'
    });

    wrapper.instance().clearInputs();

    expect(wrapper.state()).toEqual(expected);
  });

  it('should call addUser and clearInputs when submitUser is called', () => {
    wrapper.instance().clearInputs = jest.fn();

    wrapper.instance().submitUser();

    expect(mockAddUser).toHaveBeenCalled();
    expect(wrapper.instance().clearInputs).toHaveBeenCalled();
  })

})
