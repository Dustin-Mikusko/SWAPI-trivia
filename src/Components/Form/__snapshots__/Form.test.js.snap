// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form should match the snapshot with all the data passed in correctly 1`] = `
<form>
  <img
    alt="Star Wars logo"
    src="star-wars-logo.jpeg"
  />
  <input
    className="name"
    name="name"
    onChange={[Function]}
    placeholder=" Name"
    type="text"
    value=""
  />
  <input
    className="quote"
    name="quote"
    onChange={[Function]}
    placeholder=" Favorite Star Wars Quote"
    type="text"
    value=""
  />
  <select
    className="rank"
    name="rank"
    onChange={[Function]}
    value=""
  >
    <option
      value=""
    >
      --Choose Level--
    </option>
    <option
      value="beginner"
    >
      What Is Star Wars??
    </option>
    <option
      value="intermediate"
    >
      Padawan
    </option>
    <option
      value="expert"
    >
      The Force is Strong With This One...
    </option>
  </select>
  <button
    onClick={[Function]}
    type="button"
  >
    Enter, You Will
  </button>
</form>
`;
